fftrack_feb22.Rmd

```diff
@@ -214,23 +228,60 @@ grid.arrange(b_q, b_e, e_q)
 # Step 2: for each year and source_name, compute destination_fraction
 # group_by(year, source_name) %>% mutate(destination_fraction = source_quantity / sum(source_quantity))
 
-test <- output %>%
+destination <- output_slice %>%
   filter(source_name == "earth_c") %>%
   mutate(source_quantity = pool_value * source_fraction) %>%
   group_by(year, run_number) %>%
   mutate(destination_percent = source_quantity / sum(source_quantity),
-         destination_fraction = destination_percent / 100)
+         destination_fraction = destination_percent / 100) %>% 
+  ungroup()
 
 # Just looking at particular runs
-test15 <- filter(test, run_number %in% c(1:5))
+dest_runs <- filter(destination, run_number %in% c(1:5))
 
-ggplot(test15, aes(year, destination_percent, 
-                 fill = pool_name)) +
+ggplot(dest_runs, aes(year, destination_percent, 
+                   fill = pool_name)) +
   geom_area() +
   scale_fill_viridis_d() +
   facet_wrap(~run_number) +
   labs(x = "Year",
        y = "Destination fraction",
        title = "Where carbon ends up from the source earth_c")
 
-```
\ No newline at end of file
+```
+
+## What controls how much earth_c ends up in the atmosphere?
+
+```{r relative-importance}
+# Filter to just atmosphere pool, and remove unneeded columns
+dest_e2a <- filter(destination, pool_name == "atmos_c")
+dest_e2a_minimal <- dest_e2a[c("year", "destination_fraction", names(name_vector))]
+
+# Function to calculate relative importance of parameters in
+# explaining how much destination_fraction varies for a given
+# year of data. This uses relaimpo::calc.relimp()
+# See https://www.r-bloggers.com/2012/08/the-relative-importance-of-predictors-let-the-games-begin/
+# for a useful primer, along with the calc.reimp help page
+calc_relimp <- function(x) {
+  yr <- x$year[1] # record year...
+  x$year <- NULL # and drop year column
+  # Fit model. The formula notation means "destination fraction as a
+  # function of all other terms" (which is why we dropped year)
+  m <- lm(destination_fraction ~ ., data = x)
+  # Calculate relative importance metrics and extract 'lmg' results
+  # 'lmg' is the R^2 contribution averaged over orderings among regressors - should sum to one
+  relimp <- calc.relimp(m, type = "lmg", rela = TRUE)@lmg
+  # Return a data frame: year, parameter, relative importance
+  tibble(year = yr, param = names(relimp), value = relimp)
+}
+
+# Run calc_relimp above on each year's data
+dest_split <- split(dest_e2a_minimal, dest_e2a_minimal$year)
+relimp_out <- lapply(dest_split, FUN = calc_relimp) %>% 
+  bind_rows()
+
+# ...and plot!
+ggplot(relimp_out, aes(year, value, fill = param)) + 
```

Comment:
Very cool! I think I'll want to discuss the science "why" behind the graph, but conceptually this all makes sense.